internal/gitserver/git_command.go

```diff
@@ -168,35 +168,36 @@ type RemoteGitCommand struct {
 	stdin          []byte
 	noTimeout      bool
 	exitStatus     int
-	execFn         func(ctx context.Context, repo api.RepoName, op string, payload any) (resp *http.Response, err error)
+	execer         execer
+}
+
+type execer interface {
+	httpPost(ctx context.Context, repo api.RepoName, op string, payload any) (resp *http.Response, err error)
+	AddrForRepo(ctx context.Context, repo api.RepoName) (string, error)
 }
 
 // DividedOutput runs the command and returns its standard output and standard error.
 func (c *RemoteGitCommand) DividedOutput(ctx context.Context) ([]byte, []byte, error) {
-	rc, trailer, err := c.sendExec(ctx)
+	rc, err := c.sendExec(ctx)
 	if err != nil {
 		return nil, nil, err
 	}
+	defer rc.Close()
 
 	stdout, err := io.ReadAll(rc)
 	if err != nil {
+		if v := (&CommandStatusError{}); errors.As(err, &v) {
+			c.exitStatus = int(v.StatusCode)
+			if v.Message != "" {
+				return stdout, []byte(v.Stderr), errors.New(v.Message)
+			} else {
+				return stdout, []byte(v.Stderr), nil
```

Comment:
I think this issue exited with the old code path too, but I think it's a bit strange to swallow the error entirely here if v.Message is empty (even though this may not happen in practice w/ the current implementation) 


```suggestion
				return stdout, []byte(v.Stderr), v
```